File: frontend/src/api/evidence.ts
import { apiClient } from './client';
import type { EvidenceUploadResponse, EvidenceStatusResponse, EvidenceListItem } from '../types/api';

export const evidenceApi = {
  async uploadEvidence(
    caseId: string,
    file: File,
    acquisitionMethod = 'FIELD_COLLECTION',
    acquisitionContext = ''
  ): Promise<EvidenceUploadResponse> {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('acquisition_method', acquisitionMethod);
    formData.append('acquisition_context', acquisitionContext);

    const response = await apiClient.post<EvidenceUploadResponse>(
      `/cases/${caseId}/evidence/upload`,
      formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      }
    );
    return response.data;
  },

  async listEvidence(caseId: string): Promise<EvidenceListItem[]> {
    const response = await apiClient.get<EvidenceListItem[]>(`/cases/${caseId}/evidence`);
    return response.data;
  },

  async getEvidenceStatus(caseId: string, artifactId: string): Promise<EvidenceStatusResponse> {
    const response = await apiClient.get<EvidenceStatusResponse>(`/cases/${caseId}/evidence/${artifactId}`);
    return response.data;
  },

  async triggerProcessing(caseId: string, artifactId: string): Promise<{ artifact_id: string; processing_status: string; message: string }> {
    const response = await apiClient.post(`/cases/${caseId}/evidence/${artifactId}/process`);
    return response.data;
  }
};

File: civix_api/models/evidence.py
"""
CIVIX 2.0 — Evidence Pydantic Models
Round 2A

These are the request/response models for the evidence ingestion and
retrieval API. They do not contain business logic.
"""
from typing import Optional, Any, Dict
from pydantic import BaseModel
from datetime import datetime
from uuid import UUID


# ---------------------------------------------------------------------------
# Upload Response
# Returned immediately (202) when a file is accepted.
# ---------------------------------------------------------------------------
class EvidenceUploadResponse(BaseModel):
    artifact_id: UUID
    instance_id: UUID
    original_filename: str
    mime_type: str
    sha256_hash: str          # hex string representation
    file_size_bytes: int
    processing_status: str    # STORED
    is_duplicate: bool        # True if artifact already existed (same hash)
    message: str


# ---------------------------------------------------------------------------
# Evidence Status Response
# Returned by GET /evidence/{artifact_id}
# ---------------------------------------------------------------------------
class EvidenceStatusResponse(BaseModel):
    artifact_id: UUID
    instance_id: UUID
    original_filename: Optional[str]
    mime_type: Optional[str]
    file_size_bytes: Optional[int]
    processing_status: str
    processed_at: Optional[datetime]
    processing_error: Optional[str]
    media_metadata: Optional[Dict[str, Any]]
    case_id: UUID
    acquired_by: Optional[UUID]
    acquisition_method: Optional[str]
    created_at: datetime


# ---------------------------------------------------------------------------
# Evidence List Item (lightweight)
# ---------------------------------------------------------------------------
class EvidenceListItem(BaseModel):
    artifact_id: UUID
    instance_id: UUID
    original_filename: Optional[str]
    mime_type: Optional[str]
    file_size_bytes: Optional[int]
    processing_status: str
    created_at: datetime
    evidence_type: Optional[str] = None
    evidence_title: Optional[str] = None


# ---------------------------------------------------------------------------
# Process Trigger Response
# Returned by POST /evidence/{artifact_id}/process
# ---------------------------------------------------------------------------
class ProcessTriggerResponse(BaseModel):
    artifact_id: UUID
    processing_status: str
    message: str
